packages/core/test/actions.test.ts

```diff
@@ -485,6 +485,37 @@ describe('entry/exit actions', () => {
         newLightMachine.transition('red', 'NOTHING').actions.map((a) => a.type)
       ).toEqual(['exit_walk', 'exit_red', 'enter_red', 'enter_walk']);
     });
+
+    it('should exit deep descendant during a self-transition', () => {
```

Comment:
```suggestion
    it('should exit deep descendant during an external self-transition', () => {
```

Reply:
This somewhat doesn't matter because:
> The behavior of transitions with 'type' of "internal" is identical, except in the case of a transition whose source state is a compound state and whose target(s) is a descendant of the source. In such a case, an internal transition will not exit and re-enter its source state, while an external one will, as shown in the example below.

In this case, the target is not a descendant of the source and thus the behavior is identical for both external and internal transitions.